R/models.R

```diff
@@ -0,0 +1,76 @@
+cloudml_model_exists <- function(gcloud, name) {
+
+  arguments <- (MLArgumentsBuilder(gcloud)
+                ("models")
+                ("list"))
+
+  output <- gcloud_exec(args = arguments())
+  pasted <- paste(output$stdout, collapse = "\n")
+
+  suppressWarnings({
+    output <- readr::read_table2(pasted)
+  })
+
+  name %in% output$NAME
+}
+
+#' Deploy SavedModel to CloudML
+#'
+#' Deploys a SavedModel to CloudML model for online predictions.
+#'
+#' @inheritParams gcloud_config
+#'
+#' @param export_dir_base A string containing a directory containing an
+#'   exported SavedModels. Consider using \code{tensorflow::export_savedmodel()}
+#'   to export this SavedModel.
+#' @param name The name for this model. Defaults to the current directory
+#'   name.
+#' @param version The version for this model. Versions start with a letter and
+#'   contain only letters, numbers and underscores. Defaults to the current
+#'   directory name.
+#' @param cloudml A list, \code{YAML} or \code{JSON} configuration file as described
+#'   \url{https://cloud.google.com/ml-engine/reference/rest/v1/projects.jobs}.
+#'
+#' @export
+cloudml_deploy <- function(
+  export_dir_base,
+  name =  NULL,
+  version = NULL,
```

Comment:
How does it select among the potentially many versions within an `export_dir_base` (does it automatically know to pick the most recent one?)

Reply:
The CLI rather expects a path to the `saved_model.pb` file.

We could consider changing the name of the `export_dir_base` parameter, but is consistent as it is with all the `export_savedmodel()` implementations.

The error we get when not using a path to the folder containing the `.pb` file is the following:

```
 Error: ERROR: gcloud invocation failed [exit status 1]

[command]
/Users/javierluraschi/google-cloud-sdk/bin/gcloud --account 'javier@rstudio.com' --project rstudio-cloudml ml-engine versions create cloudml3 --model=cloudml --origin=../tfserve/trained/keras-mnist/ --staging-bucket=gs://rstudio-cloudml --runtime-version=1.4

[output]


[errmsg]
ERROR: (gcloud.ml-engine.versions.create) FAILED_PRECONDITION: Field: version.deployment_uri Error: Deployment directory gs://rstudio-cloudml/93e72607fbefd6e2ad10a1073e3e514a93ea2e657d6a0b3550293064fe7a6e57/ is expected to contain exactly one of: [saved_model.pb, saved_model.pbtxt].
- '@type': type.googleapis.com/google.rpc.BadRequest
  fieldViolations:
  - description: 'Deployment directory gs://rstudio-cloudml/93e72607fbefd6e2ad10a1073e3e514a93ea2e657d6a0b3550293064fe7a6e57/
      is expected to contain exactly one of: [saved_model.pb, saved_model.pbtxt].'
    field: version.deployment_uri 
```

Reply:
I agree we should use `export_dir_base`, however within an `export_dir_base` I've noticed there are multiple subdirectories each of which includes a distinct version of the model. My question is how we select from among these versions when `version = NULL`. It would be nice if we could automatically choose the most recent one but I'm not sure if that's appropriate / practical.